src/AzureAuth/CommandMain.cs

```diff
@@ -218,6 +222,46 @@ public static string PrefixedPromptHint(string promptHint)
             return $"{PromptHintPrefix}: {promptHint}";
         }
 
+        /// <summary>
+        /// Generates event data from the AuthFlowResult.
+        /// </summary>
+        /// <param name="result">The AuthFlowResult.</param>
+        /// <returns>The event data.</returns>
+        public EventData GenerateEventData(AuthFlowResult result)
+        {
+            if (result == null)
+            {
+                return null;
+            }
+
+            var eventData = new EventData();
+            eventData.Add("authflow", result.AuthFlowName);
+            eventData.Add("success", result.Success);
+
+            var correlationIDs = new List<string>();
+
+            if (result.Errors.Any())
+            {
+                var error_messages = JsonSerializer.Serialize(result.Errors.Select(error => ExceptionsExtensions.ToFormattedString(error)));
+                eventData.Add("error_messages", error_messages);
+                correlationIDs = ExceptionListToStringConverter.ExtractCorrelationIDsFromException(result.Errors);
+            }
+
+            if (result.Success)
+            {
+                correlationIDs.Add(result.TokenResult.CorrelationID.ToString());
+                eventData.Add("token_validity_hours", result.TokenResult.ValidFor.Hours);

```

Comment:
```suggestion
                eventData.Add("token_valid_for_minutes", result.TokenResult.ValidFor.TotalMinutes);

```

Reply:
Need to use the `TotalMinutes` otherwise it doesn't compute the entire time span into that unit.